### PR TITLE
Add tick state machine and clock reset animation

### PR DESCRIPTION
**Pull request by karnkaul**
Press `R` or `Enter` to reset clock to initial state.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -44,13 +36,50 @@ class input_t {
...
+			f32 delta{};
+			if (m_input->any_held(key_t::Down, key_t::Left, key_t::S, key_t::A)) { delta += -1.0f; }
+			if (m_input->any_held(key_t::Up, key_t::Right, key_t::W, key_t::D)) { delta += 1.0f; }
+			add(translation_scale * delta * dt.count());
+		}
+	}
+
+  private:
+	using flags_t = u8;
+	enum flag : flags_t { none = 0, zeroing = 1 << 0 };
```
Comment (karnkaul) [inline src/main.cpp:65]: This allows stuffing up to 8 flags into a single byte, whereas `bool`s would occupy a byte each. Another approach would be to force single bit storage via `u8 zeroing : 1`, but I prefer this.

File: src/main.cpp [hunk 2]
```diff
@@ -44,13 +36,50 @@ class input_t {
...
+
+	bool test(flags_t f) const noexcept { return (flags & f) == f; }
+	void update(flags_t set, flags_t unset = 0) noexcept {
+		flags |= set;
+		flags &= ~unset;
+	}
+
+	void add(hour_t offset) noexcept {
+		(*m_clock) += offset;
+		m_offset += offset;
```
Comment (karnkaul) [inline src/main.cpp:80]: Since there is no "zero point" on the clock itself, this class to figure it out via its own offset.

File: src/main.cpp [hunk 3]
```diff
@@ -44,13 +36,50 @@ class input_t {
...
-	out_clock += scale * delta * dt.count();
-}
+class clock_ticker_t {
+  public:
+	clock_ticker_t(input_t const* input, world_clock_t* clock) noexcept : m_input(input), m_clock(clock) {}
+
+	void operator()(sec_t dt) noexcept {
+		if (m_input->any_held(key_t::Enter, key_t::R)) { update(flag::zeroing); }
+		static constexpr f32 translation_scale = 10.0f;
+		if (test(flag::zeroing)) {
```
Comment (karnkaul) [inline src/main.cpp:46]: Gist of this branch: keep subtracting `delta` until less than that remains, then get rid of the residue and reset the flag.

File: src/main.cpp [hunk 4]
```diff
@@ -63,9 +92,10 @@ int main() {
 	std::cout << clock << '\n';
 	input_t input;
 	delta_time_t dt;
+	clock_ticker_t tick{&input, &clock};
 	while (ctx.running()) {
 		input.update(ctx.poll());
-		tick(clock, input, ++dt);
+		tick(++dt);
```
Comment (karnkaul) [inline src/main.cpp:98]: Masquerading as a function call 😎

**Review by hentai-chan — APPROVED** [review]

**Review by NikolaJelic — APPROVED** [review]

**Pull request merged**